[BugFix] Fix some compatibility for auto increment column (#18443)

srlch opened the pull request:
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #18443

## Problem Summary(Required) ：
Problem:
1. Some new metadata persisted in BackupMeta which incompatible with older versions.
2. Auto-increment param in OlapTableSink will be init uncorrectly when FE is older versions.
3. remove Auto Increment Id map when drop database will cause dead lock when FE restart and replay dropDb. 

Solution:
1. persist the new metadata for backup in BackupJobInfo, which is JSON stored in remote but not in Image File.
2. Check Fe version when init param in OlapTableSink.
3. use isReplay to skip saving remove auto increment id editlog.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

srlch commented:
run starrocks_be_unittest

srlch commented:
run starrocks_fe_unittest

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 17 (52.94%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/BackupJobInfo.java|9|17|52.94%|[172, 174, 175, 177, 326, 327, 414, 418]|

ABingHuang commented on fe/fe-core/src/main/java/com/starrocks/backup/BackupJobInfo.java:
```diff
@@ -388,6 +410,12 @@ private static void genFromJson(String json, BackupJobInfo jobInfo) {
+            try {
+                tblInfo.autoIncrementId = tbl.getLong("autoIncrementId");
+            } catch (Exception e) {
+                LOG.info("read from older BackJobInfo.");
```
```suggestion
                LOG.warn("read from older BackJobInfo.", e);
```

srlch replied:
done

srlch committed:
[BugFix] Fix some compatibility for auto increment column (#18443)

Problem:
1. Some new metadata persisted in BackupMeta which incompatible with older versions.
2. Auto-increment param in OlapTableSink will be init uncorrectly when FE is older versions.
3. remove Auto Increment Id map when drop database will cause dead lock when FE restart and replay dropDb

Solution:
1. persist the new metadata for backup in BackupJobInfo, which is JSON stored in remote but not in Image File.
2. Check Fe version when init param in OlapTableSink.
3. use isReplay to skip saving remove auto increment id editlog.

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 26 (61.54%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|2|00.00%|[494, 606]|
|:large_blue_circle:|com/starrocks/backup/BackupJobInfo.java|9|17|52.94%|[172, 174, 175, 177, 326, 327, 414, 418]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
@Mergifyio backport branch-3.0
